cli/generate/build.rs

```diff
@@ -0,0 +1,33 @@
+use std::{env, path::PathBuf, process::Command};
+
+fn main() {
+    if let Some(git_sha) = read_git_sha() {
+        println!("cargo:rustc-env=BUILD_SHA={git_sha}");
+    }
+}
+
+// This is copied from the build.rs in parent directory. This should be updated if the
+// parent build.rs gets fixes.
+fn read_git_sha() -> Option<String> {
+    let crate_path = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+
+    if !crate_path
+        .parent()
+        .unwrap()
```

Comment:
This will panic if there is no parent.

Reply:
Added a `Some(crate_path.parent())?;` check before to return early. Should work, but unsure if it's idioamtic rust or whatever.